File: Klanvision_backend/src/main/java/com/klanvision/backend/model/AuditActivity.java
package com.klanvision.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "activities")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AuditActivity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String user;
    private String action;
    private String type;
    private String status;
    
    @Column(columnDefinition = "TEXT")
    private String details;
    
    private LocalDateTime timestamp = LocalDateTime.now();
}

File: Klanvision_V5_Backend-master/models/AuditActivity.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/db.js';

const AuditActivity = sequelize.define('AuditActivity', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  user: {
    type: DataTypes.STRING
  },
  action: {
    type: DataTypes.STRING
  },
  type: {
    type: DataTypes.STRING
  },
  status: {
    type: DataTypes.STRING
  },
  details: {
    type: DataTypes.TEXT
  },
  timestamp: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  }
}, {
  tableName: 'activities',
  timestamps: false
});

export default AuditActivity;
export { AuditActivity };
